code/modules/turbolift/turbolift.dm

```diff
@@ -0,0 +1,266 @@
+// This is area based! Ensure each turbolift is in a unique area, or things will get fucky																									//
+// Ensure that turbolift doors are at least one tile away from the next lift. See DeepSpace13.dmm for examples. 																			//
+// Use the indestructible elevator turfs or it'll look terrible!				  																											//
+// Modify pixel_x and y as needed, it starts off snapped to the tile below a wall.																											//
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+//Areas//
+
+/area/turbolift //Only use subtypes of this area
+	requires_power = FALSE //no APCS in the lifts please
+	ambientsounds = list('sound/effects/turbolift/elevatormusic.ogg')
+
+/area/turbolift/primary
+	name = "primary turbolift"
+
+/area/turbolift/secondary
+	name = "secondary turbolift"
+
+/area/turbolift/tertiary
+	name = "tertiary turbolift"
+
+
+
+//Structures and logic//
+
+/obj/structure/turbolift
+	name = "turbolift control console"
+	icon = 'icons/obj/turbolift.dmi'
+	icon_state = "lift-off"
+	density = FALSE
+	anchored = TRUE
+	can_be_unanchored = FALSE
+	mouse_over_pointer = MOUSE_HAND_POINTER
+	desc = "Nanotrasen's decision to replace the iconic turboladder was not met with unanimous praise, experts citing increased obesity figures from crewmen no longer needing to climb vertically through several miles of deck to reach their target. However this is undoubtedly much faster."
+	var/floor_directory = "<font color=blue>\
+		Deck 1: Engineering <br>\
+		Deck 2: Promenade<br>\
+		Deck 3: Bridge<br>\
+		</font>" //Change this if you intend to make a new map. Helps players know where they're going.
+	var/list/turbolift_turfs = list()
+	var/floor = 0 //This gets assigned on init(). Allows us to calculate where the lift needs to go next.
+	var/list/destinations = list() //Any elevator that's on our path.
+	pixel_y = 32 //This just makes it easier for locate...
+	var/in_use = FALSE //Is someone using a lift? If they are, then don't let anyone in the lift.
+	var/max_floor = 0 //Highest floor we can go to?
+	var/bolted = FALSE //Is this door bolted or unbolted? do we need to change that if a person walks into our lift?
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	var/is_controller = FALSE //Are we controlling the other lifts?
+	var/obj/structure/turbolift/master = null //Who is the one pulling the strings
+	var/obj/machinery/door/airlock/linked_door = null //Linked elevator door
+
+/obj/machinery/door/airlock/turbolift
+	name = "turbolift airlock"
+	icon = 'icons/obj/turbolift_door.dmi'
+	desc = "A sleek airlock for walking through. This one looks extremely strong."
+	icon_state = "closed"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
```

Comment:
Why indestructible

Reply:
Because making elevators work at all is hard enough. Having them be constructable/deconstructable is a lot harder.

Reply:
Making them deconstructable is really easy. Making them constructible is a lot harder and probably doesn't need to be done.

Reply:
I'd prefer if permanently breaking the elevators wasn't a thing.